cmd/sqlcmd/sqlcmd.go

```diff
@@ -118,23 +120,172 @@ func (a SQLCmdArguments) authenticationMethod(hasPassword bool) string {
 }
 
 func Execute(version string) {
-	ctx := kong.Parse(&args, kong.NoDefaultHelp())
-	if args.Version {
-		ctx.Printf("%v\n", version)
-		fmt.Println(localizer.Sprintf("Legal docs and information: aka.ms/SqlcmdLegal"))
-		fmt.Println(localizer.Sprintf("Third party notices: aka.ms/SqlcmdNotices"))
-		os.Exit(0)
+	// Cobra Tryout #2 : This has version and help flags. But they are also getting stucked inside run function without error
+	rootCmd := &cobra.Command{
+		Use:   "testCommand",
+		Short: "A brief description of your command",
+		PersistentPreRunE: func(cmd *cobra.Command, argss []string) error {
```

Comment:
>PersistentPreRunE [](http://example.com/codeflow?start=2&length=17)

why do we have both PersistentPreRunE and PreRunE?

Reply:
Its by mistake while testing each . Removing PreRunE.